### PR TITLE
basics of metrics

### PR DESCRIPTION
**Pull request by kennethreitz**

### CODE REVIEW

**Comment by kennethreitz** [comment]
work in progress

**Comment by kennethreitz** [comment]
This also adds a new warnings check for inclusion of the six library, which is needed with the release of the latest setuptools chain, which has a few dependencies, unfortunately. 

Plan to add the other dependencies as well, eventually, but six is the one people run into the most. 

I want to look into removing these entries from `requirements.txt` automatically.

File: bin/utils [hunk 1]
```diff
@@ -1,5 +1,12 @@
+#!/usr/bin/env bash
 shopt -s extglob
 
+# The standard library.
+if [[ ! -f  /tmp/stdlib.sh ]]; then
+  curl -s https://raw.githubusercontent.com/heroku/buildpack-stdlib/v2/stdlib.sh > /tmp/stdlib.sh
```
Comment (jmorrell) [inline bin/utils]: Add retry flags `--retry 5 --retry-max-time 15`

File: bin/utils [hunk 2]
```diff
@@ -108,3 +74,8 @@ sub-env() {
   )
 }
 
+# Measure the size of the Python installation.
+measure-size() {
```
Comment (jmorrell) [inline bin/utils]: The name is pretty generic for being so specific. `measure-python-directory`?

File: bin/warnings [hunk 3]
```diff
@@ -37,6 +40,18 @@ distribute-included() {
     puts-warn "This library is automatically installed by Heroku and shouldn't be in"
     puts-warn "Your requirements.txt file. This can cause unexpected behavior."
     puts-warn "  -- Much Love, Heroku."
+    mcount 'warnings.distribute'
+  fi
+}
+
+six-included() {
```
Comment (jmorrell) [inline bin/warnings]: This should probably be a different PR, but we can roll with it

File: bin/utils (hunk 2 above)
Comment (kennethreitz) [inline bin/utils]: it's the exact same name as the one used in the node buildpack

Comment (jmorrell) [inline bin/utils]: Then I'll update that one :)

File: bin/compile [hunk 4]
```diff
@@ -137,10 +141,14 @@ if [[ $BUILD_DIR != '/app' ]]; then
 fi
 
 # Install Python.
+let start=$(nowms)
```
Comment (kennethreitz) [inline bin/compile]: this is the documented way of doing this.

File: bin/utils (hunk 1 above)
Comment (kennethreitz) [inline bin/utils]: good call
